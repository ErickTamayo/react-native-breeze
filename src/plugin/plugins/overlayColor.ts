import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = { key: string };

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^overlay-(?<key>${keys("overlayColor")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, color }) => {
  const { key } = groups;
  const value = color("overlayColor", key);
  return { overlayColor: value };
};
