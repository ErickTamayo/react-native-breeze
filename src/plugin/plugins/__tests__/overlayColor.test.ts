import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../overlayColor";

describe("overlayColor", () => {
  const shouldMatch = ["overlay-green-500", "overlay-black"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["overlay-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
