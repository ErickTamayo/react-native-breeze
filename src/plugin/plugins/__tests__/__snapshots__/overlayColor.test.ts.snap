// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlayColor Should parse (overlay-black) correctly 1`] = `
Object {
  "overlayColor": "#000000",
}
`;

exports[`overlayColor Should parse (overlay-green-500) correctly 1`] = `
Object {
  "overlayColor": "#48bb78",
}
`;
